app/src/main/java/com/dragic/gamehunter/viewmodel/HomeViewModel.kt

```diff
@@ -19,12 +20,27 @@ class HomeViewModel @Inject constructor(
     var dealData by mutableStateOf<List<DealViewState>>(emptyList())
         private set
 
+    private var pageNumber by mutableIntStateOf(1)
```

Comment:
We should only use `mutableStateOf` when we want Compose to react to changes of some value. 

In this case, `pageNumber` is private, so Compose doesn't see it. We also don't need Compose to react to changes of `pageNumber`, it should only react to changes of `dealData` by displaying a new list.

Because of this, you can just replace this line with:

```kt
private var pageNumber = 1
```